core/src/apps/bitshares/base/helpers.py

```diff
@@ -0,0 +1,75 @@
+from trezor import wire
+from trezor.crypto import base58
+
+from apps.common import HARDENED
+
+
+def base58_encode(prefix: str, data: bytes) -> str:
+    return "{}{}".format(prefix, base58.encode(data))
+
+
+def base58_decode(prefix: str, data: str) -> bytes:
+    data = data.replace(prefix, "", 1)
```

Comment:
Uff .. that's not cool.
Instead, use
```
data = data[len(prefix):]
```
Else you may change the Public key itself!

Reply:
Will change that. But normally it can only change the public key when there is no prefix in which case the public key would already be wrong.